packages/cli/src/tests/utils/ast.js

```diff
@@ -0,0 +1,179 @@
+          should(countOcurrances(result, 'searches:')).eql(1);
+
+          const codeByLine = result.split('\n').map(x => x.trim());
+          const firstIndex = codeByLine.indexOf('searches: {');
+          // assertions about what comes in the trigger property
+          should(codeByLine.indexOf('[findThing.key]: findThing')).eql(
+            firstIndex + 1
+          );
+          should(codeByLine[firstIndex + 2]).eql('}');
+        });
+      });
+    });
+
+    describe('legacy apps', () => {
+      it('should add a property to an existing action type', () => {
+        const result = addKeyToPropertyOnApp(
+          sampleLegacyAppIndex,
+          'triggers',
+          'getThing'
+        );
+        should(countOcurrances(result, 'triggers:')).eql(2);
+        should(countOcurrances(result, 'searches:')).eql(2);
+
+        const codeByLine = result.split('\n').map(x => x.trim());
+
+        // find the second occurance, the one that's not in the "legacy" property
+        const operativeIndex = codeByLine.indexOf(
+          'triggers: {',
+          codeByLine.indexOf('triggers: {') + 1
+        );
+
+        should(
+          codeByLine.indexOf('[businessTrigger.key]: businessTrigger,')
+        ).eql(operativeIndex + 1);
+        should(codeByLine.indexOf('[getThing.key]: getThing')).eql(
+          operativeIndex + 2
+        );
+        should(codeByLine[operativeIndex + 3]).eql('},');
+      });
+
+      it('should add a property to an existing empty action type', () => {
+        const result = addKeyToPropertyOnApp(
+          sampleLegacyAppIndex,
+          'searches',
+          'findThing'
+        );
+        should(countOcurrances(result, 'searches:')).eql(2);
+
+        const codeByLine = result.split('\n').map(x => x.trim());
+        // find the second occurance, the one that's not in the "legacy" property
+        const operativeIndex = codeByLine.indexOf('searches: {');
+        should(codeByLine.indexOf('[findThing.key]: findThing')).eql(
+          operativeIndex + 1
+        );
+        should(codeByLine[operativeIndex + 2]).eql('},');
+      });
+
+      it('should add a new property if action type is missing', () => {
+        const result = addKeyToPropertyOnApp(
+          sampleLegacyAppIndex,
+          'resources',
+          'findThing'
+        );
+        should(countOcurrances(result, 'triggers:')).eql(2);
+        should(countOcurrances(result, 'searches:')).eql(2);
+        should(countOcurrances(result, 'resources:')).eql(1);
+
+        const codeByLine = result.split('\n').map(x => x.trim());
+        const firstIndex = codeByLine.indexOf('resources: {');
+        // assertions about what comes in the trigger property
+        should(codeByLine.indexOf('[findThing.key]: findThing')).eql(
+          firstIndex + 1
+        );
+        should(codeByLine[firstIndex + 2]).eql('}');
+      });
+    });
+  });
+
+  describe('errors', () => {});
```

Comment:
I'll wrap this up tomorrow, should be straightforward. I'll have an index.js that specifically will be tripped over by each thrown error.